File: 2195-time-needed-to-buy-tickets/time-needed-to-buy-tickets.go
func timeRequiredToBuy(tickets []int, k int) int {
    n := len(tickets)
    time:=0
    if k != n-1 {
        time = k+1
        for i := 0 ; i<=k ; i++ {
            tickets[i]--
        }
    }

    
    for i:=0;i<n;i++ {
        if tickets[i] <= tickets[k]{
            time += tickets[i]
        } else {
            time += tickets[k]
        }
    }
    return time
}

File: 2195-time-needed-to-buy-tickets/time-needed-to-buy-tickets.py
class Solution:
    def timeRequiredToBuy(self, tickets: List[int], k: int) -> int:
        ans = k+1
        for i in range(k+1):
            tickets[i]-=1
        # print(tickets)
        buy = tickets[k]
        if buy == 0 :
            return ans
        for ticket in tickets:
            ans += min(ticket,buy)
        
        return ans